webrtc.html

```diff
@@ -3073,44 +3073,54 @@ <h4>
+                        <ul>
+                          <li><var>parsedURL</var> is failure</li>
+                          <li><var>parsedURL</var>'s [=url/scheme=] is neither `"stun"`,
+                          `"stuns"`, `"turn"`, nor `"turns"`</li>
+                          <li><var>parsedURL</var> does not have an [=url/opaque path=]</li>
+                          <li><var>parsedURL</var>'s' [=url/fragment=] is non-null</li>
+                          <li><var>parsedURL</var>'s' [=url/scheme=] is `"stun"` or `"stuns"`,
+                          and <var>parsedURL</var>'s' [=url/query=] is non-null</li>
+                        </ul>
+                      </li>
+                      <li>
+                        <p>If <var>parsedURL</var>'s [=url/scheme=] is not implemented by the
+                        user agent, then [=exception/throw=] a {{NotSupportedError}}.</p>
+                      </li>
+                      <li>
+                        <p>Let <var>hostAndPortURL</var> be result of
+                        <a data-cite="!url#concept-url-parser">parsing</a> the concatenation of
+                        `"https://"` and <var>parsedURL</var>'s [=url/path=].</p>
+                      </li>
+                      <li>
+                        <p>If <var>hostAndPortURL</var> is failure, then [=exception/throw=] a
+                        "{{SyntaxError}}" {{DOMException}}.</p>
+                        <p class="note">For "stun" and "stuns" schemes, this validates
+                        [[!RFC7064]] section 3.1.<br>
+                        For "turn" and "turns" schemes, this and the steps below validate
+                        [[!RFC7065]] section 3.1.</p>
+                      </li>
+                      <li>
+                        <p>If <var>parsedURL</var>'s [=url/query=] is non-null, run the following
+                        sub-steps:</p>
+                        <ol>
+                          <li><p>TODO: validate ?transport=udp|tcp</p></li>
+                        </ol>
                       </li>
                       <li>
                         <p>
-                          If <var>scheme name</var> is <code class=
-                          "scheme">turn</code> or <code class=
-                          "scheme">turns</code>, and
-                          <var>server</var>.{{RTCIceServer/credential}} is not
-                          a <span class="idlMemberType">DOMString</span>, then
-                          [= exception/throw =] an {{InvalidAccessError}}.
+                          If <var>parsedURL</var>'s' [=url/scheme=] is <code>"turn"</code> or
+                          <code>"turns"</code>, and neither
+                          <var>server</var>.{{RTCIceServer/username}} nor
+                          <var>server</var>.{{RTCIceServer/credential}}
+                          [=map/exist=], then [= exception/throw =] an {{InvalidAccessError}}.
```

Comment:
I don't think it helps to have credential without username. The previous text insisted that both need to exist.

Reply:
Whoops, I inverted the logic incorrectly. Thanks for spotting that.